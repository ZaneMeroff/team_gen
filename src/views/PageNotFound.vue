<template>
  <div class="page-not-found-container">

    <p class="err-txt">sorry, this page was not found :(</p>

    <router-link class="router-link" to="/">
      <button>home</button>
    </router-link>

  </div>
</template>

<script>

  export default {
    name: "PageNotFound",
  }
</script>

<style scoped>

  .page-not-found-container {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .err-txt {
    font-size: 1rem;
    margin-top: 50px;
  }

  button {
    height: 40px;
    width: 80px;
    border-radius: 20px;
    border: 2px solid #000;
    margin: 30px 0px;
  }

  button:hover {
    cursor: pointer;
  }

</style>
