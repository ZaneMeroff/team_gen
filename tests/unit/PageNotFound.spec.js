import { createLocalVue, shallowMount } from "@vue/test-utils"
import PageNotFound from "@/views/PageNotFound"
import VueRouter from "vue-router"

// --------- router setup ------------

const localVue = createLocalVue()
localVue.use(VueRouter)

// -----------------------------------

describe("PageNotFound", () => {

  describe("snapshots", () => {

    it("should render with default html", () => {
      const component = shallowMount(PageNotFound, { localVue })

      expect(component.element).toMatchSnapshot()
    })
  })
})
