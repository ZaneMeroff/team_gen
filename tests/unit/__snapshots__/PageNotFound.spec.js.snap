// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound snapshots should render with default html 1`] = `
<div
  class="page-not-found-container"
>
  <p
    class="err-txt"
  >
    sorry, this page was not found :(
  </p>
   
  <router-link-stub
    ariacurrentvalue="page"
    class="router-link"
    event="click"
    tag="a"
    to="/"
  >
    <button>
      home
    </button>
  </router-link-stub>
</div>
`;
